### PR TITLE
Create settings activity

### PR DESCRIPTION
**Pull request by abadaracco**
This branch contains more than only the settings activity:
- We updated the app's flow, as now we have an splash screen, a landing page and a settings page.
-  We implemented the settings activity using the Preference class from android to have the look of a classic android settings page. And as a result of this we used the default shared preferences to save the settings.
- We implemented the logic of sending messages with some stopping criteria and using the MessageController.
- The functionality of executing the app in background using the android's AlarmManager was also implemented.
- And also a lot of small functionality improvements were made

### CODE REVIEW

File: app/src/main/java/uy/com/bix/app/smsproject/activity/MainActivity.java [hunk 1]
```diff
@@ -24,10 +25,10 @@ protected void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
 		setContentView(R.layout.activity_main);
 
-		connectUi();
+		//connectUi();
 	}
 
-	private void connectUi() {
+	/*private void connectUi() {
```
Comment (acabreragnz) [inline app/src/main/java/uy/com/bix/app/smsproject/activity/MainActivity.java]: Why do you keep commented code?

File: app/src/main/java/uy/com/bix/app/smsproject/activity/SettingsActivity.java [hunk 2]
```diff
@@ -1,18 +1,200 @@
...
+
+		JodaTimeAndroid.init(this);
+
+		// Get saved user settings
+		SharedPreferences settings = PreferenceManager.getDefaultSharedPreferences(contextOfApplication);
+
+		// The month in the date picker is in 0-11 range
+		expirationMonth = settings.getInt("Month", 1) - 1;
+		expirationDay = settings.getInt("Day", 1);
+		expirationHour = settings.getInt("Hour", 23);
```
Comment (acabreragnz) [inline app/src/main/java/uy/com/bix/app/smsproject/activity/SettingsActivity.java]: **In the future** will be great to have constants for this kind of values.

For example:

`private static int DEFAULT_HOUR = 23;`

Or better in a configuration file.

File: app/src/main/java/uy/com/bix/app/smsproject/activity/SettingsActivity.java [hunk 3]
```diff
@@ -1,18 +1,200 @@
...
+		expirationMinute = settings.getInt("Minute", 30);
+		isLastDay = settings.getBoolean("LastDay", false);
+		expirationYear = settings.getInt("Year", 2016);
```
Comment (acabreragnz) [inline app/src/main/java/uy/com/bix/app/smsproject/activity/SettingsActivity.java]: What does 2016 mean?

File: app/src/main/java/uy/com/bix/app/smsproject/activity/SettingsActivity.java [hunk 4]
```diff
@@ -1,18 +1,200 @@
...
+	@Override
+	protected void onDestroy() {
+		super.onDestroy();
+		getDelegate().onDestroy();
+	}
+
+	@Override
+	protected void onResume() {
+		super.onResume();
+		// Set up a listener whenever a key changes
```
Comment (acabreragnz) [inline app/src/main/java/uy/com/bix/app/smsproject/activity/SettingsActivity.java:178]: Only as a comment, I think is clear to put an empty line before comments.

File: app/src/main/java/uy/com/bix/app/smsproject/activity/SettingsActivity.java [hunk 5]
```diff
@@ -1,18 +1,200 @@
...
+		getPreferenceScreen().getSharedPreferences()
+			.registerOnSharedPreferenceChangeListener(this);
+	}
+
+	@Override
+	protected void onPause() {
+		super.onPause();
+		// Unregister the listener whenever a key changes
+		getPreferenceScreen().getSharedPreferences()
```
Comment (acabreragnz) [inline app/src/main/java/uy/com/bix/app/smsproject/activity/SettingsActivity.java]: I don't know if you are following a style guide of coding but 

the more comfortable way to indent this kind of things I think is:

```
getPreferenceScreen()
    .getSharedPreferences()
    .unregisterOnSharedPreferenceChangeListener(this);
```

or

```
getPreferenceScreen()
.getSharedPreferences()
.unregisterOnSharedPreferenceChangeListener(this);
```

File: app/src/main/java/uy/com/bix/app/smsproject/activity/SettingsActivity.java [hunk 6]
```diff
@@ -23,17 +205,146 @@ public boolean onCreateOptionsMenu(Menu menu) {
...
+		sendingDate = sendingDate.withMonthOfYear(expirationMonth + 1);
+		sendingDate = sendingDate.withDayOfMonth(expirationDay);
+
+		// The date must be in milliseconds
+		long whenToFireTask = sendingDate.getMillis();
+
+		isActive = settings.getBoolean("Active", false);
+		if (isActive) {
+			scheduleAlarm(whenToFireTask);
+			System.out.println(sendingDate);
```
Comment (acabreragnz) [inline app/src/main/java/uy/com/bix/app/smsproject/activity/SettingsActivity.java:288]: `System.out.println(sendingDate);`

File: app/src/main/java/uy/com/bix/app/smsproject/controllers/AlarmController.java [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+			// We need to update the preferneces of the user according to the month
+			SharedPreferences.Editor editor = settings.edit();
+			editor.putInt("Year", nextExpirationDate.getYear());
+			editor.putInt("Month", nextExpirationDate.getMonthOfYear());
+			editor.putInt("Day", nextExpirationDate.getDayOfMonth());
+			editor.apply();
+
+			// The date must be in milliseconds
+			dateOfNextFiring = nextExpirationDate.getMillis();
+			System.out.println("Esta Fecha:");
```
Comment (acabreragnz) [inline app/src/main/java/uy/com/bix/app/smsproject/controllers/AlarmController.java]: `System.out.println("Esta Fecha:");`

And the others sout

File: app/src/main/java/uy/com/bix/app/smsproject/controllers/AlarmController.java [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+	private AlarmController() {}
+
+	public static AlarmController getInstance() {
+		if (instance == null) {
+			instance = new AlarmController();
+		}
+		return instance;
+	}
+
+	public void prepareAndSendMessage (Context context) {
```
Comment (acabreragnz) [inline app/src/main/java/uy/com/bix/app/smsproject/controllers/AlarmController.java]: what do you mean with "prepare"?
how do you prepare a message?

Here could be a nice javadoc :)

File: app/src/main/java/uy/com/bix/app/smsproject/controllers/AlarmController.java [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+		SharedPreferences settings = PreferenceManager.getDefaultSharedPreferences(context);
+		JodaTimeAndroid.init(context);
+		isActive = settings.getBoolean("Active", false);
+
+		// We make sure the app is active to send messages
+		if (isActive) {
+
+			// We get all the information the user set
+			notifyWhenSending = settings.getBoolean("Notify", false);
```
Comment (acabreragnz) [inline app/src/main/java/uy/com/bix/app/smsproject/controllers/AlarmController.java:109]: Take in mind for the future, I would create a function for this block and other for 57-62. 

> Functions only have to do one thing

File: app/src/main/java/uy/com/bix/app/smsproject/controllers/AlarmController.java [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+
+		// If the day changes we stop sending messages to prevent spending wrong credit
+		DateTime actualTime = DateTime.now();
+		boolean stopForChangeOfDay = actualTime.getDayOfMonth() != expirationDate.getDayOfMonth();
+
+		while (!stopForMax && !stopForChangeOfDay) {
+			msgController.sendMessage(phoneNum, text, context);
+
+			// We update the stopping criteria
+			maxMessagesToSend --;
```
Comment (acabreragnz) [inline app/src/main/java/uy/com/bix/app/smsproject/controllers/AlarmController.java]: `maxMessagesToSend--;`

File: app/src/main/java/uy/com/bix/app/smsproject/controllers/AlarmController.java [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+			stopForMax = maxMessagesToSend == 0;
```
Comment (acabreragnz) [inline app/src/main/java/uy/com/bix/app/smsproject/controllers/AlarmController.java:147]: nice

File: app/src/main/java/uy/com/bix/app/smsproject/controllers/AlarmController.java [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+			actualTime = DateTime.now();
+			stopForChangeOfDay = actualTime.getDayOfMonth() != expirationDate.getDayOfMonth();
```
Comment (acabreragnz) [inline app/src/main/java/uy/com/bix/app/smsproject/controllers/AlarmController.java:149]: nice nice

**Comment by acabreragnz** [comment]
Amazing Job! :dancers:

File: app/src/main/java/uy/com/bix/app/smsproject/activity/SettingsActivity.java (hunk 6 above)
Comment (abadaracco) [inline app/src/main/java/uy/com/bix/app/smsproject/activity/SettingsActivity.java:288]: The system out are left in purpose to debug and test the app, they'll be deleted in the future

File: app/src/main/java/uy/com/bix/app/smsproject/controllers/AlarmController.java (hunk 7 above)
Comment (abadaracco) [inline app/src/main/java/uy/com/bix/app/smsproject/controllers/AlarmController.java]: The system out are left in purpose to debug and test the app, they'll be deleted in the future

File: app/src/main/java/uy/com/bix/app/smsproject/controllers/AlarmController.java (hunk 8 above)
Comment (abadaracco) [inline app/src/main/java/uy/com/bix/app/smsproject/controllers/AlarmController.java]: Excellent comment and advice, thanks!

**Pull request merged**